### ISSUE TITLE
Evaluation mode

### ISSUE THREAD

**Issue by lianrzh** [NTMC-Community/MatchZoo-py#101]
Do you think that the evaluation function would be needed to set `model.eval()`?

**Comment by Chriskuei** [comment]
When evaluating, the model has already been set in evaluation mode.
https://github.com/NTMC-Community/MatchZoo-py/blob/2d274879f444bb2556503d59145f7fd61652cea9/matchzoo/trainers/trainer.py#L341-L353

**Comment by lianrzh** [comment]
@Chriskuei 
`def evaluate(
        self,
        dataloader: DataLoader,
    ):
        """
        Evaluate the model.
        :param dataloader: A DataLoader object to iterate over the data.
        """
        result = dict()
        y_pred = self.predict(dataloader)
        y_true = dataloader.label
        id_left = dataloader.id_left

        if isinstance(self._task, tasks.Classification):
            for metric in self._task.metrics:
                result[metric] = metric(y_true, y_pred)
        else:
            for metric in self._task.metrics:
                result[metric] = self._eval_metric_on_data_frame(
                    metric, id_left, y_true, y_pred.squeeze(axis=-1))
        return result
`
